agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1025,21 +1023,16 @@ bool backhaul_manager::backhaul_fsm_main(bool &skip_select)
                 for (const auto &socket : slaves_sockets) {
                     if (socket) {
                         for (int i = 0; i < beerocks::IFACE_TOTAL_VAPS; ++i) {
-                            if (socket->vaps_list.vaps[i].mac != network_utils::ZERO_MAC) {
+                            if ((socket->vaps_list.vaps[i].mac != network_utils::ZERO_MAC) &&
+                                (socket->vaps_list.vaps[i].ssid[0] != '\0')) {
                                 bssid_list.push_back(socket->vaps_list.vaps[i].mac);
                             }
                         }
                     }
                 }
-                // We must generate a new MID for the periodic AP Metrics Response messages that
-                // do not correspond to an AP Metrics Query message.
-                // We cannot set MID to 0 here because we must also differentiate periodic
-                // AP Metrics Response messages and messages received from monitor thread
-                // due to channel utilization crossed configured threshold value.
-                // As a temporary solution, set MID to UINT16_MAX here.
-                // TODO: to be fixed as part of #1328
```

Comment:
I am not sure I fully understand all considerations. However, there is a discussion in #1381 that might shed some light on the issue.
What I can tell is that we use `mid=0` as the mid for the repeated, internally initiated, metrics requests (as oppose to specifically requested by the controller and as oppose to those that are initiated by the monitor because of changes in the stations).
The monitor generated vs. controller generated vs repeatedly generated metrics queries is not hermetically closed - there was a long oral discussion of how to ensure that the responses that come from the different initiators are handled correctly - no bullet proof solution was found. However, current situation seems to be good enough for practical usage.

Reply:
0 means the response message will be generated with the new MID